.github/workflows/progress-tracker.yml

```
@@ -173,7 +177,7 @@ jobs:
               chapters.push({
                 id: issue.number,
                 url: issue.html_url,
-                title: heading.title || issue.title,
+                title: heading.title || issue.title.replace(/ 20[0-9][0-9]/,''),
```

Sticking the pattern to the end to make sure no other 4-digit numbers are replaced (unlikely false negatives).

```suggestion
                title: heading.title || issue.title.replace(/\s+\d{4}$/,''),
```

---

What about this instead?:

```suggestion
                title: heading.title || issue.title.replace(/\b20[0-9][0-9]\b/,''),
```

---

I think we should be very specific about the trailing year only. If we want to replace year anywhere in the title then there can be better ways. The change you proposed has a couple of issues: 1) if there are more than one occurrences of years in the title, it will replace the first one, and 2) a word boundary does not include the space character itself, so the replacement may leave spurious white space (not a big deal though).

```
> 'Review of 2020 HTTP/2 Chapter 2021'.replace(/\b20[0-9][0-9]\b/, '')
'Review of  HTTP/2 Chapter 2021'
```

```
> 'Review of 2020 HTTP/2 2021'.replace(/\s+20\d{2}$/, '')
'Review of 2020 HTTP/2'
```

---

Switched to the `\b` format and `.replace(/  /,' ').trim()` so it covers your scenarios and `2021 JavaScript` or `JavaScript 2021`.

I think we're really getting into the weeds here if we do any more here though! While it's good to future proof and make it reasonably robust, it's only for this chapter tracker which is an internal tool for us, we already have the format for this year's chapters and had to make changes to support 2021 so likely will have to make more changes next year.

I think the current code sufficiently covers us without making it needless overly complex.

---

I do not like unnecessary chaining of replacements followed by a call to trim. If the intent is to remove a trailing 4-digit number, then using `.replace(/\s+\d{4}$/,'')` is all we need. If we want to limit it to 4-digit numbers that start with `20` then `.replace(/\s+20\d{2}$/, '')` will do the trick. If we want things to be even more tight and replace only the current year then we can use `new Date().getFullYear()` as the find string.

---

None of those cover the use case of `2022 JavaScript Chapter`, which is not unlikely to be used in the future.

Also, given the frequent use of chains elsewhere in this script, often it really long, complex lines of code, it seems a little odd to complain about this relatively simple one.

---

I did not realize that we are also anticipating titles with leading year, not just trailing ones. :)

We can go for something like `.replace(new Date().getFullYear(), '').trim()`

```
> t1 = '2021 JavaScript Chapter'
> t2 = 'JavaScript Chapter 2021'
> t1.replace(new Date().getFullYear(), '').trim()
"JavaScript Chapter"
> t2.replace(new Date().getFullYear(), '').trim()
"JavaScript Chapter"
```

---

Does not handle when chapters slip into the following year (like two of them did this year), nor dates in the middle of the title which will have double spaces now - which you, yourself raised!

---

I considered those two situations, but found them to be unlikely because 2) by the time things will slip into the next year, we will already be settled with the section and chapter numbers and the issue title will not be used at that time, and 2) I do not anticipate current year to be anything but as a suffix or prefix to the title.